torch/_subclasses/fake_tensor.py

```diff
@@ -1084,8 +1116,13 @@ def _cache_key(
             # Disallowing dynamic shapes can introduce a DynamicOutputShapeException
             # where it wasn't seen on a previous instance of the same op.
             self.shape_env.settings if self.shape_env else None,
-        )
-        return _DispatchCacheKey(key_values)
+        ]
+        # Translate any FakeTensor args to metadata.
+        if args:
```

Comment:
Interesting. Did you find this actually helps? I thought I remembered playing around with avoiding calls to _prep_args_for_hash and I couldn't eek out any more perf.